test/error_messages.coffee

```diff
@@ -989,3 +995,160 @@ test "`&&=` and `||=` with a space in-between", ->
     a or = 1
          ^
   '''
+
+test "anonymous functions cannot be exported", ->
+  assertErrorFormat '''
+    export ->
+      console.log 'hello, world!'
+  ''', '''
+    [stdin]:1:8: error: unexpected ->
+    export ->
+           ^^
+  '''
+
+test "anonymous classes cannot be exported", ->
+  assertErrorFormat '''
+    export class
+      constructor: ->
+        console.log 'hello, world!'
+  ''', '''
+    SyntaxError: Unexpected token export
```

Comment:
Dammit, you’re completely correct. The generated JavaScript for that test is:
```js
export var (function() {
  function _Class() {
    console.log('hello, world!');
  }

  return _Class;

})();
```
which is completely invalid. I’ll add a check to catch this within the CoffeeScript compiler. Thanks for pointing this out.